src/server/project.ts

```diff
@@ -2504,16 +2506,29 @@ export class AutoImportProviderProject extends Project {
                 //    package and load the JS.
                 if (packageJson && compilerOptions.allowJs && compilerOptions.maxNodeModuleJsDepth) {
                     const entrypoints = getRootNamesFromPackageJson(packageJson, program, symlinkCache, /*resolveJs*/ true);
-                    rootNames = concatenate(rootNames, entrypoints);
-                    dependenciesAdded += entrypoints?.length ? 1 : 0;
+                    dependenciesAdded += addRootNames(entrypoints);
                 }
             }
         }
 
-        if (rootNames?.length) {
-            hostProject.log(`AutoImportProviderProject: found ${rootNames.length} root files in ${dependenciesAdded} dependencies in ${timestamp() - start} ms`);
+        const references = program.getResolvedProjectReferences();
+        let referencesAddded = 0;
+        if (references?.length && program.getCompilerOptions().paths) {

```

Comment:
What’s the reason for blocking this on `paths`?

Reply:
`shouldOfferImportCompletions` is ~~pretty sus tbh~~ something I’ve considered deleting in favor of just relying on the user preference to disable auto-imports. I was just thinking of examples like our own codebase, where we have project references set up, but reference into them with relative paths rather than `paths` aliases or node_modules symlinks.

Reply:
Then should I test emit module kind is not none ? Is that better ? We do need to check if project is modules or not right

Reply:
`--module none` is basically meaningless; see https://github.com/microsoft/TypeScript/issues/39603 and https://github.com/microsoft/TypeScript/issues/55256#issuecomment-1706823298

IMO disabling auto-imports for the project via settings is the way to go.

Reply:
But currently there is no such setting.. i would prefer not to merge these two things. Do i create this unconditionally for now and when we have disable auto import preference we could add it ?

Reply:
This is what I’m referring to. Does this control only VS Code’s behavior?

<img width="335" alt="image" src="https://github.com/microsoft/TypeScript/assets/3277153/545d8398-6c8f-4ea8-964c-26f2ebfd4e21">

Reply:
i dont see anything changing in tsserver log by enabling or disabling this. Not sure. Is this same as `includePackageJsonAutoImports` ? if that is it we can include by default since we have already figured out we need to create auto import provider

Reply:
Yeah but right now vscode doesnt send this as part of their initial configure. Its only in completions request..

Here is what vscode sends:
```
{
      "seq": 0,
      "type": "request",
      "command": "configure",
      "arguments": {
        "hostInfo": "vscode",
        "preferences": {
          "providePrefixAndSuffixTextForRename": true,
          "allowRenameOfImportPath": true,
          "includePackageJsonAutoImports": "auto",
          "excludeLibrarySymbolsInNavTo": true
        },
        "watchOptions": {}
      }
    }
```

Reply:
I would propose we check that UserPreference anyway as the condition here (other clients might send it?) and then update VS Code to start sending it in Configure. We can discuss that next week at the editor sync.